### PR TITLE
309: Put a regular text hyperlink under the widget that performs the desired action, 310: Energy details is showing wrong data for future block

### PR DESCRIPTION
**Pull request by Natsvora**
309: Added click events instead of hyperlinks on the action widget

### CODE REVIEW

File: frontend/src/app/oe/components/base-box-hor/base-box-hor.component.ts [hunk 1]
```diff
@@ -39,20 +41,24 @@ export class BaseBoxHorComponent implements OnInit, OnDestroy {
   ngOnDestroy(): void {
   }
 
+  navigateTo(): void {
+    !isTextSelection() && this.router.navigate([this.link]);
+  }
+
   toHHMMSS(secs) {
```
Comment (dambaev) [inline frontend/src/app/oe/components/base-box-hor/base-box-hor.component.ts]: I see basebox.component.ts contains the same function. I suggest to refactor it into reusable one.

**Review by dambaev — CHANGES_REQUESTED** [review]

**Comment by dambaev** [comment]
my suggestion was about `toHHMMSS` function, not about navigation

**Review by dambaev — DISMISSED** [review]
my suggestion was about `toHHMMSS` function, not about navigation

**Comment by girish-velas** [comment]
I am fine with the code - but would recommend creating components where same code exists to keep everything lean

**Review by girish-velas — APPROVED** [review]
try to create common components in a follow up PR to make code better

**Pull request merged**